CDS/js/youtube/youtube-upload-lib.js

```diff
@@ -0,0 +1,106 @@
+var Promise = require('promise');
+var fs = require('fs');
+var youtubeAuth = require('./youtube-auth.js');
+var googleapis = require('googleapis');
+var OAuth2 = googleapis.auth.OAuth2;
+var dataStore = require('../DataStore.js');
+
+const YOUTUBE_API_VERSION = 'v3';
+
+function getMetadata() {
+    var title, description, category, status;
+
+    if (!process.env.title) {
+        throw new Error('Cannot upload to youtube: missing a title');
+    }
+
+    title = process.env.title;
```

Comment:
this should be a string substitution.  the js lib provides all substitutions as Promises, so I'd recommend setting each up as a Promise and then doing Promise.all.andthen() or whatever